### PR TITLE
[fingerprint] add useRNCoreAutolinkingFromExpo to support new autolinking

### PR DESCRIPTION
**Pull request by Kudo**
# Why

since we have expo-modules-autolinking to support the core react-native autolinking sources, we can use it to save some time from fingerprint

# How

add `useRNCoreAutolinkingFromExpo` to support fingerprint

# Benchmark result

bare-expo: `23.048s -> 21.425s`
blank sdk51 project: `881.8ms -> 737.9ms`
react-conf-app: `4.962s -> 2.075s`

# Test Plan

add unit tests and ci passed

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Kudo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/expo/expo/31638?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#31638** <a href="https://app.graphite.dev/github/pr/expo/expo/31638?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#31637** <a href="https://app.graphite.dev/github/pr/expo/expo/31637?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Kudo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/@expo/fingerprint/CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,13 @@
 
 ### 🛠 Breaking changes
 
+- This version includes fingerprint result breaking changes. ([#31638](https://github.com/expo/expo/pull/31638) by [@kudo](https://github.com/kudo))
```
Comment (wschurman) [inline packages/@expo/fingerprint/CHANGELOG.md:7]: I wonder if there's a way to indicate that every version bump is a breaking change. Might be simpler for this library.

**Review by wschurman — APPROVED** [review]

File: packages/@expo/fingerprint/CHANGELOG.md (hunk 1 above)
Comment (Kudo) [reply packages/@expo/fingerprint/CHANGELOG.md:7]: so far it's still maintainable and i still want to keep patch version bump as non-breaking change. having the changelog helps the person doing package publish to understand whether it should be a major/minor/patch version bump.
if maintaining breaking changes comes much hassle in the future, we could clarify a better approach for this. thanks!

**Comment by Kudo** [comment]
rebase to resolve changelog conflict

**Pull request merged**